docs/modules/default-architecture/pages/generation.adoc

```diff
@@ -0,0 +1,51 @@
+= Generation
+
+Prompt engineering is a reflection of your thinking.
+
+LLMs will do what you tell them to. It's up to you to start with a clear idea of the problem you want the LLM to solve, so you can provide everything it needs to do what it does well.
```

Comment:
Suggestions - 

Prompts are the starting points for the LLM generation process. They can provide context, set the tone, define objectives, and ultimately shape the quality of the response. Prompt design is an entire field of its own, and we recommend [further reading of this to ...]. However, we [will] lay out a starting prompt for a basic RAG pipeline, and show a technique (CO-STAR) for organizing prompts for more complex tasks.  

Problem: You have proprietary knowledge you want to accurately [and with direct examples] impart to readers. You ...

...
...
...

The additional prompting defines context and new objectives for the LLM. However, [just like humans], LLMs can get confused if the prompt is poorly organized or contains non-relevant text. 

[Following is my personal recommendation based off of a very limited set of reading on prompt engineering]

A simple way to improve quality of generation is to follow the CO-STAR framework, a template that sections out key aspects that influence response generation. 

[Following acronym list is from https://towardsdatascience.com/how-i-won-singapores-gpt-4-prompt-engineering-competition-34c195a93d41#10b2]

(C) Context: Provide background information on the task
(O) Objective: Define what the task is that you want the LLM to perform
(S) Style: Specify the writing style you want the LLM to use
(T) Tone: Set the attitude of the response
(A) Audience: Identify who the response is intended for
(R) Response: Provide the response format

Using this framework, we can organize our previous prompt.
```
prompt_template = """ 
# CONTEXT # 
You are a tool called IRL Company Chatbot. You are a technical expert with a specific knowledge base supplied to you via the context. 

# OBJECTIVE # 
* Answer questions based only on the given context. 
* If possible, include reference URLs in the following format: add "https://docs.irl.ai/docs" before the "slug" value of the document. For any URL references that start with "doc:" or "ref:" use its value to create a URL by adding "https://docs.irl.ai/docs/" before that value. For reference URLs about release notes add "https://docs.irl.ai/changelog/" before the "slug" value of the document. Do not use page titles to create urls. 
* If the answer cannot be found in the documentation, write "I could not find an answer. Join our [Slack Community](https://www.irl.ai/slackinvite) for further clarifications." 
* Do not make up an answer or give an answer that is not supported by the provided context.

# STYLE # 
Follow the writing style of technical experts. 

# TONE # 
Professional

# AUDIENCE # 
People that want to learn about IRL Company. 

# RESPONSE # 
The response should be in the following format:
---
answer

url_reference
---

Context: {context}
Question: {question}
Your answer:
""" 
```

Reply:
Could make the response format more interesting, then show an example response to display better how the response can be drastically changed based on the prompt.

Reply:
Added